web/tests/typeahead_helper.test.js

```diff
@@ -529,7 +529,7 @@ test("sort_recipients dup alls", () => {
         current_topic: "Linux topic",
     });
 
-    const expected = [all_obj, all_obj, a_user];
```

Comment:
In this test, we have
```
 // full_name starts with same character but emails are 'all'
 const test_objs = [all_obj, a_user, all_obj];
```

I think we can't have users with email as "all".  We use "all" email for `@all` wildcard only.

So what this test is actually trying to verify; some historic reason?

Also, we can't have duplicate `@all` object after filtering as per the [current code logic](https://github.com/zulip/zulip/blob/main/web/src/composebox_typeahead.js#L499).

ref: [test created](https://github.com/zulip/zulip/commit/d2279885195cd35f51835916f55eed8b55d07890#diff-f3ec133ee89eb97671a4342214f21097925b846aecce24a4eb1dabc656bf858dR369)

Reply:
I think it must be that it was intended for a case where a user's name was "all"? I don't have another theory; it's probably just a bad test.